editor/src/components/canvas/canvas-strategies/strategies/paste-metastrategy.tsx

```diff
@@ -0,0 +1,247 @@
+              new Set(existingIDs),
+            ).value
+
+            const reparentTarget: StaticReparentTarget =
+              strategy === 'REPARENT_AS_ABSOLUTE'
+                ? {
+                    strategy: strategy,
+                    insertionPath: target.parentPath,
+                    intendedCoordinates: absolutePositionForPaste(
+                      target,
+                      currentValue.originalElementPath,
+                      {
+                        originalTargetMetadata:
+                          elementPasteWithMetadata.targetOriginalContextMetadata,
+                        currentMetadata: canvasState.startingMetadata,
+                        originalPathTrees:
+                          interactionSession.interactionData.targetOriginalPathTrees,
+                        currentPathTrees: canvasState.startingElementPathTree,
+                      },
+                      interactionSession.interactionData.canvasViewportCenter,
+                    ),
+                  }
+                : { strategy: strategy, insertionPath: target.parentPath }
+
+            const indexPosition =
+              target.type === 'sibling'
+                ? absolute(
+                    MetadataUtils.getIndexInParent(
+                      editor.jsxMetadata,
+                      editor.elementPathTree,
+                      target.siblingPath,
+                    ) + 1,
+                  )
+                : front()
+
+            const result = insertWithReparentStrategies(
+              editor,
+              elementPasteWithMetadata.targetOriginalContextMetadata,
+              interactionSession.interactionData.targetOriginalPathTrees,
+              reparentTarget,
+              {
+                elementPath: currentValue.originalElementPath,
+                pathToReparent: elementToReparent(elementWithUniqueUID, currentValue.importsToAdd),
+              },
+              indexPosition,
+              canvasState.builtInDependencies,
+            )
+
+            if (result == null) {
+              return []
+            }
+
+            return foldAndApplyCommandsInner(
+              editor,
+              [],
+              [
+                ...result.commands,
+                updateSelectedViews('always', [...editor.selectedViews, result.newPath]),
+              ],
+              commandLifecycle,
+            ).statePatches
+          }),
+        ]
+      })
+
+      if (commands.length === 0) {
+        return emptyStrategyApplicationResult
+      }
+
+      return strategyApplicationResult([updateSelectedViews('always', []), ...commands])
+    },
+  }
+}
+
+export const pasteMetaStrategy =
+  (mode: PasteMode) =>
+  (
+    canvasState: InteractionCanvasState,
+    interactionSession: InteractionSession | null,
+  ): CanvasStrategy | null => {
+    if (interactionSession?.interactionData.type !== 'STATIC_REPARENT') {
+      return null
+    }
+
+    return mode === 'replace'
```

Comment:
I suggest to use inner or even anonymous functions and not embedded ternaries, they are so hard to parse.

Reply:
fixed in https://github.com/concrete-utopia/utopia/pull/3798/commits/0db863999a814ca1e76a95bcd52206264c386256